### PR TITLE
Facilitate flaky test resolution

### PR DESCRIPTION
**Pull request by pellared**
## Why

Checking if a test is flaky is not user-friendly.

## What

- Add `TestName` Nuke parameter to enable running a single .NET test (unit + integration)
- Add `TestCount` Nuke parameter to enable running .NET tests (unit + integration) multiple times
- Add `verify-test` GH workflow that can be used to check if a given test is flaky

## Tests

The new GH workflow:
- Linux - `SqlClientTests`: https://github.com/pellared/opentelemetry-dotnet-instrumentation/actions/runs/3059880715
- Windows - `AspNetTests`: https://github.com/pellared/opentelemetry-dotnet-instrumentation/actions/runs/3059881973
- All: - `SmokeTests`: https://github.com/pellared/opentelemetry-dotnet-instrumentation/actions/runs/3059882944

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: .github/workflows/verify-test.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            6.0.x
+
+      - name: Windows - build and test ${{ github.event.inputs.testName }} ${{ github.event.inputs.count }} times
+        run: |
+          ./build.cmd BuildTracer ManagedTests --skip RunManagedTests
+          if ($global:LASTEXITCODE -ne 0) { 
+            exit $global:LASTEXITCODE
+          }
+          
+          for ($i=1; $i -le ${{ github.event.inputs.count }}; $i++) {
```
Comment (pellared) [inline .github/workflows/verify-test.yml:36]: I am considering adding a `TestCount` nuke parameter instead.
What is your preference?

Comment (Kielek) [reply .github/workflows/verify-test.yml:36]: It could be useful for local testing. I like the idea.

Comment (pellared) [reply .github/workflows/verify-test.yml]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1229/commits/7a4284731aeca05dcea8710fc6184c62c699b6b6

**Review by Kielek — APPROVED** [review]

**Review by pjanotti — COMMENTED** [review]
Nice. We really need to improve in this regard.

**Pull request merged**